src/main/java/openblocks/common/BeepGenerator.java

```diff
@@ -95,8 +95,17 @@ public void run() {
 	}
 
 	private void writeSample(SourceDataLine line) {
-		byte[] buffer = generateSamplesWithSweep(this.lastToneFrequency, this.toneFrequency);
-		this.lastToneFrequency = this.toneFrequency;
+		double lastToneFrequency = this.toneFrequency;
+		if (this.toneFrequency == 0d || this.targetToneFrequency == 0d) {
+			this.toneFrequency = targetToneFrequency;
+			lastToneFrequency = targetToneFrequency;
+		}
+		else if (this.toneFrequency < targetToneFrequency)
+			this.toneFrequency += Math.min(50d, targetToneFrequency - this.toneFrequency);
```

Comment:
jep, but since you generate 100ms at once instead of 16ms, so that would be 31,25 if I'm correct.
After some I used 50 as a "compromise", since it's a little easier to fly that way.